### PR TITLE
feat(vg_lite): add radial gradient support

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Description of the feature or fix

image

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by kisvegabor** [comment]
Sooner or later we shall figure out how to describe radial an conic gradient in styles too.

**Review by kisvegabor — COMMENTED** [review]
Can we add tests via the ThorVG simulator?

**Comment by FASTSHIFT** [comment]
> Can we add tests via the ThorVG simulator?

Yes, we can consider joining thorvg's test, but it should be noted that the results produced by rendering may not be guaranteed to be completely consistent with software rendering, and independent reference images may need to be re-generated.

**Comment by kisvegabor** [comment]
With a test we could at lesz check if VG-Lite really processes the radial gradient settings. 

>  but it should be noted that the results produced by rendering may not be guaranteed to be completely consistent with software rendering

It's not an issue. We are not testing how VG-Lite renders, but only that if it really executes what it is asked for. 

 > and independent reference images may need to be re-generated.

How do you mean it?

**Comment by FASTSHIFT** [comment]
> With a test we could at lesz check if VG-Lite really processes the radial gradient settings.
> 
> > but it should be noted that the results produced by rendering may not be guaranteed to be completely consistent with software rendering
> 
> It's not an issue. We are not testing how VG-Lite renders, but only that if it really executes what it is asked for.
> 
> > and independent reference images may need to be re-generated.

OK.

> How do you mean it?

What I mean is that I need to generate an independent reference image for vg-lite. This should not be a problem.

**Comment by FASTSHIFT** [comment]
I can send a PR later for vg-lite test, and this PR can be merged first~

File: Kconfig [hunk 1]
```diff
@@ -335,6 +335,20 @@ menu "LVGL configuration"
 				which usually improves performance,
 				but does not guarantee the same rendering quality as the software.
 
+		config LV_VG_LITE_LINEAER_GRAD_CACHE_SIZE
```
Comment (XuNeo) [inline Kconfig]: Let's use the name of `CNT` instead of `SIZE`, similar with header cache.

Comment (FASTSHIFT) [reply Kconfig]: Done.

**Comment by FASTSHIFT** [comment]
> I can send a PR later for vg-lite test, and this PR can be merged first~

@kisvegabor Do you agree? 😄

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]
Agreed, merged :slightly_smiling_face:

**Pull request merged**